server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/SetDefaultActionTest.java

```diff
@@ -53,75 +54,122 @@
-  private DbClient dbClient = db.getDbClient();
-  private QProfileWsSupport wsSupport = new QProfileWsSupport(dbClient, userSessionRule, defaultOrganizationProvider);
+  private WsActionTester tester;
+  private DefaultOrganizationProvider defaultOrganizationProvider;
+  private DbClient dbClient;
+  private QProfileWsSupport wsSupport;
+  private OrganizationDto organization;
+  private SetDefaultAction underTest;
+
+  /** Single, default quality profile for language xoo1 */
+  private QualityProfileDto xoo1Profile;
+  /** Parent quality profile for language xoo2 (not a default) */
+  private QualityProfileDto xoo2Profile;
+  /** Child quality profile for language xoo2, set as default */
+  private QualityProfileDto xoo2Profile2;
 
   @Before
   public void setUp() {
-    createProfiles();
+    defaultOrganizationProvider = TestDefaultOrganizationProvider.from(db);
+    dbClient = db.getDbClient();
+    wsSupport = new QProfileWsSupport(dbClient, userSessionRule, defaultOrganizationProvider);
+    organization = OrganizationTesting.newOrganizationDto();
+    db.organizations().insert(organization);
+    underTest = new SetDefaultAction(LanguageTesting.newLanguages(xoo1Key, xoo2Key), dbClient, userSessionRule, wsSupport);
+
+    String organizationUuid = organization.getUuid();
+    xoo1Profile = QualityProfileTesting.newQualityProfileDto()
+      .setOrganizationUuid(organizationUuid)
+      .setLanguage(xoo1Key)
+      .setDefault(true);
+    xoo2Profile = QualityProfileTesting.newQualityProfileDto()
+      .setOrganizationUuid(organizationUuid)
+      .setLanguage(xoo2Key);
+    xoo2Profile2 = QualityProfileTesting.newQualityProfileDto()
+      .setOrganizationUuid(organizationUuid)
+      .setLanguage(xoo2Key)
+      .setParentKee(xoo2Profile.getKee())
+      .setDefault(true);
+    dbClient.qualityProfileDao().insert(db.getSession(), xoo1Profile, xoo2Profile, xoo2Profile2);
+    db.commit();
 
-    tester = new WsTester(new QProfilesWs(
-      mock(RuleActivationActions.class),
-      mock(BulkRuleActivationActions.class),
-      new SetDefaultAction(LanguageTesting.newLanguages(xoo1Key, xoo2Key), new QProfileLookup(dbClient), new QProfileFactory(dbClient), wsSupport)));
+    tester = new WsActionTester(underTest);
   }
 
   @Test
   public void set_default_profile_using_key() throws Exception {
     logInAsQProfileAdministrator();
 
-    checkDefaultProfile(xoo1Key, "sonar-way-xoo1-12345");
-    checkDefaultProfile(xoo2Key, "my-sonar-way-xoo2-34567");
+    checkDefaultProfile(xoo1Key, xoo1Profile.getKey());
+    checkDefaultProfile(xoo2Key, xoo2Profile2.getKey());
+
+    TestResponse response = tester.newRequest()
+      .setMethod("POST")
+      .setParam("profileKey", xoo2Profile.getKee()).execute();
 
-    tester.newPostRequest("api/qualityprofiles", "set_default").setParam("profileKey", "sonar-way-xoo2-23456").execute().assertNoContent();
+    assertThat(response.getInput()).isEmpty();
```

Comment:
@simonbrandhof made this comment on a PR of mine and I think it makes sense, you could check the response code (200) in addition to the fact that the response is empty